### ISSUE TITLE
incorrect "reportUnusedVariable" is detected in version 1.1.91

### ISSUE THREAD

**Issue by yamap55** [microsoft/pyright#1224]
**Describe the bug**
The "reportUnusedVariable", which did not occur in version 1.1.90, now occurs in version 1.1.91.

**To Reproduce**
Performing an inspection with the following codes will cause an error.

**Expected behavior**
not error.

**Screenshots or Code**

```python
from typing import Dict, List
def func(list_in_dict: List[Dict]):
    _d = {}
    return [_d(**d) for d in list_in_dict]
```

**VS Code extension or command-line**
command-line
version: 1.1.91

**Additional context**

- execute command

```
$ sudo npm install --global pyright@1.1.90
/usr/bin/pyright -> /usr/lib/node_modules/pyright/index.js
/usr/bin/pyright-langserver -> /usr/lib/node_modules/pyright/langserver.index.js
+ pyright@1.1.90
updated 1 package in 0.996s
$ pyright --version
pyright 1.1.90
$ pyright /hoge/huga.py
Loading configuration file at /hoge/pyrightconfig.json
Auto-excluding **/node_modules
Auto-excluding **/__pycache__
Auto-excluding .git
stubPath /hoge/typings is not a valid directory.
Searching for source files
Found 1 source file
0 errors, 0 warnings, 0 infos 
Completed in 0.742sec
$ sudo npm install --global pyright@1.1.91
/usr/bin/pyright -> /usr/lib/node_modules/pyright/index.js
/usr/bin/pyright-langserver -> /usr/lib/node_modules/pyright/langserver.index.js
+ pyright@1.1.91
updated 1 package in 1.016s
$ pyright --version
pyright 1.1.91
$ pyright /hoge/huga.py
Loading configuration file at /hoge/pyrightconfig.json
Auto-excluding **/node_modules
Auto-excluding **/__pycache__
Auto-excluding .git
stubPath /hoge/typings is not a valid directory.
Searching for source files
Found 1 source file
/hoge/huga.py
  6:25 - error: Variable "d" is not accessed (reportUnusedVariable)
1 error, 0 warnings, 0 infos 
Completed in 0.727sec
```

**Comment by erictraut** [comment]
In the example you provided `_d` is a dictionary and is not callable. Do you have `reportGeneralTypeIssues` disabled (to suppress the "is not callable" error) but `reportUnusedVariables` enabled?

**Comment by erictraut** [comment]
There was a regression introduced in the latest version that causes `d` to be reported as unaccessed because of the other error in the code. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is now fixed in Pyright 1.1.92, which I just published. It will also be addressed in next week's Pylance release.
